contribute.md

```diff
@@ -3,104 +3,345 @@ layout: contribute
+
+One way is to find Rails version by opening the [Gemfile](https://github.com/theforeman/foreman/blob/develop/Gemfile) and searching for `gem 'rails'` line, then finding out which Ruby versions are supported for that particular Ruby on Rails version. Another, perhaps faster way, is to visit our [Jenkins CI server](http://ci.theforeman.org/view/Foreman%20pipeline/job/test_develop) to see matrix of supported versions. Then pick any version from that list.
 
-Any version of Ruby that is older than those listed is not supported.
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf module enable ruby:2.7
+dnf install ruby ruby-devel
+```
 
 #### Supported Node.js versions
-In general, the latest stable Node.js should work. However if you are just starting out, you may want to develop against
-one of the versions that we test against to reduce the chances of you hitting an unexpected issue, although this is not
-required.
 
-You can see what versions we are currently testing against on our Travis CI runs in the .travis.yml file in the foreman
-repository.
+In general, the latest stable Node.js should work. However if you are just starting out, you may want to develop against one of the versions that we test against to reduce the chances of you hitting an unexpected issue, although this is not required.
 
-Any version of Node.js that is older than those listed is not supported.
+You can view our [Github Actions configuration](https://github.com/theforeman/foreman/blob/develop/.github/workflows/js_tests.yml) to see supported NodeJS versions. Any version from that list will do. For populating the node_modules folder, npm version 4 or later is required. Alternatively, [yarn](https://yarnpkg.com) can also be used.
 
-For populating the node_modules folder, npm version 4 or later is required. Alternatively,
-[yarn](https://yarnpkg.com) can also be used.
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
 
-#### Setup development environment
-* Fork [theforeman/foreman](https://github.com/theforeman/foreman) to your GitHub account. This will create a "foreman" repo under your GitHub username.
-* Clone the fork you just created to your development system:
-{% highlight bash %}
-git clone https://github.com/<username>/foreman.git
-{% endhighlight %}
-* Reference theforeman/foreman as upstream:
-{% highlight bash %}
+```
+dnf module enable nodejs:14
+dnf install nodejs
+```
+
+#### Supported PostgreSQL versions
+
+In general, the latest version of PostgreSQL will work. Minimum version is currently 10, anything newer than that will do the job.
+
+Instructions for Fedora, CentOS 8, CentOS 8 Stream or Red Hat Enterprise Linux:
+
+```
+dnf module enable postgresql:10
```

Comment:
Why would you enable version 10 specifically? I think it's safe to drop this line. On Fedora version 10 may even be dropped. I'd either recommend the default (10 on EL8) or actually choose version 12 since that's what we deploy on EL7 and recommend in our upgrade guide (but don't automate yet).

So either:
```suggestion
```

Or
```suggestion
dnf module enable postgresql:12
```

Reply:
Okay.